project-minimization-core/src/main/kotlin/org/plan/research/minimization/core/algorithm/dd/impl/ProbabilisticDD.kt

```diff
@@ -95,4 +97,40 @@ class ProbabilisticDD : DDAlgorithm {
         currentItems.addAll(buffer)
         buffer.clear()
     }
+
+    private fun <T> initProbs(
+        items: List<T>,
+        info: (T) -> DDItemInfo,
+    ): IdentityHashMap<T, Double> {
+        val probs = IdentityHashMap<T, Double>()
+        val important = IdentityHashMap<T, Boolean>()
+        items.forEach { important[it] = info(it).likelyImportant }
+
+        val importantItemsCount = items.count { important[it]!! }
+        if (importantItemsCount == 0 || importantItemsCount == items.size) {
+            val bisectProb = splitAtProb(items.size / 2.0)
+            items.forEach { probs[it] = bisectProb }
+            return probs
+        }
+
+        val unimportantItemsCount = items.size - importantItemsCount
+        val unimportantProb = splitAtProb(unimportantItemsCount.toDouble())
+        val importantProb = max(
+            splitAtProb(importantItemsCount.toDouble() / 2),
+            unimportantProb.pow(0.9),
```

Comment:
Let's use something like prob if the next operation is unsuccessful (seems p/(1 - (1 - p)^n), n = unimportant, p = prob of unimportant). WDYT?

Reply:
So that in such case all items have equal weight in the next step? Sounds okay